### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by EdwinRozario**
Organised the logic of the gem into 2 major Classes and moved common methods to a Helper module. New ruby syntaxes updated. All the old specs are organised in different files.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
...
 require './lib/sad_panda'
 require 'lingua/stemmer'
-require './lib/sad_panda/emotions/emotion_bank'
-require './lib/sad_panda/emotions/term_polarities'
-require './lib/sad_panda/emotions/stopwords'
+require './lib/sad_panda/bank/emotions'
+require './lib/sad_panda/bank/polarities'
+require './lib/sad_panda/bank/stopwords'
+
+require 'pry'
```
Comment (mattThousand) [inline spec/spec_helper.rb]: @EdwinRozario we probably don't need `pry` outside of the debugging/development stage. Let's remove it here

Comment (EdwinRozario) [reply spec/spec_helper.rb]: Done!

**Comment by EdwinRozario** [comment]
I have made the change you have requested. But i havent updated the version number yet.

**Comment by mattThousand** [comment]
Hey @EdwinRozario, thank you for all of this work! Sorry for being delinquent (getting wrapped up in a `Swift` project).

Running the test suite on your branch, I am getting a few test failures. Are you able to address them? It's been quite a while since I've worked in this codebase, so I may be speaking without proper context. Let me know what you think, I've pasted my `RSpec` output below

```Randomized with seed 50061
.......................FFFFF.....................................FF.....

Failures:

  1) SadPanda#polarity returns a Float value polarity
     Failure/Error: polarities.empty? ? 5.0 : (polarities.sum / polarities.length)

     NoMethodError:
       undefined method `sum' for [10.0]:Array
     # ./lib/sad_panda/polarity.rb:21:in `call'
     # ./lib/sad_panda.rb:22:in `polarity'
     # ./spec/sad_panda_spec.rb:65:in `block (3 levels) in <top (required)>'
     # ./spec/sad_panda_spec.rb:68:in `block (3 levels) in <top (required)>'

  2) SadPanda#polarity returns 8.0 for sad emoji
     Failure/Error: polarities.empty? ? 5.0 : (polarities.sum / polarities.length)

     NoMethodError:
       undefined method `sum' for [2.0]:Array
     # ./lib/sad_panda/polarity.rb:21:in `call'
     # ./lib/sad_panda.rb:22:in `polarity'
     # ./spec/sad_panda_spec.rb:80:in `block (3 levels) in <top (required)>'

  3) SadPanda#polarity returns 10.0
     Failure/Error: polarities.empty? ? 5.0 : (polarities.sum / polarities.length)

     NoMethodError:
       undefined method `sum' for [10.0]:Array
     # ./lib/sad_panda/polarity.rb:21:in `call'
     # ./lib/sad_panda.rb:22:in `polarity'
     # ./spec/sad_panda_spec.rb:65:in `block (3 levels) in <top (required)>'
     # ./spec/sad_panda_spec.rb:72:in `block (3 levels) in <top (required)>'

  4) SadPanda#polarity returns 5.0 for happy & sademoji
     Failure/Error: polarities.empty? ? 5.0 : (polarities.sum / polarities.length)

     NoMethodError:
       undefined method `sum' for [5.0, 8.0, 2.0]:Array
     # ./lib/sad_panda/polarity.rb:21:in `call'
     # ./lib/sad_panda.rb:22:in `polarity'
     # ./spec/sad_panda_spec.rb:84:in `block (3 levels) in <top (required)>'

  5) SadPanda#polarity returns 8.0 for happy emoji
     Failure/Error: polarities.empty? ? 5.0 : (polarities.sum / polarities.length)

     NoMethodError:
       undefined method `sum' for [8.0]:Array
     # ./lib/sad_panda/polarity.rb:21:in `call'
     # ./lib/sad_panda.rb:22:in `polarity'
     # ./spec/sad_panda_spec.rb:76:in `block (3 levels) in <top (required)>'

  6) SadPanda::Polarity#call when input contains a word with polarity returns 10.0
     Failure/Error: polarities.empty? ? 5.0 : (polarities.sum / polarities.length)

     NoMethodError:
       undefined method `sum' for [10.0]:Array
     # ./lib/sad_panda/polarity.rb:21:in `call'
     # ./spec/sad_panda/polarity_spec.rb:35:in `block (4 levels) in <top (required)>'

  7) SadPanda::Polarity#call when input contains a word with polarity adds polarities of words to polarities Array
     Failure/Error: polarities.empty? ? 5.0 : (polarities.sum / polarities.length)

     NoMethodError:
       undefined method `sum' for [10.0]:Array
     # ./lib/sad_panda/polarity.rb:21:in `call'
     # ./spec/sad_panda/polarity_spec.rb:39:in `block (4 levels) in <top (required)>'

Finished in 0.05021 seconds (files took 0.17354 seconds to load)
72 examples, 7 failures

Failed examples:

rspec ./spec/sad_panda_spec.rb:67 # SadPanda#polarity returns a Float value polarity
rspec ./spec/sad_panda_spec.rb:79 # SadPanda#polarity returns 8.0 for sad emoji
rspec ./spec/sad_panda_spec.rb:71 # SadPanda#polarity returns 10.0
rspec ./spec/sad_panda_spec.rb:83 # SadPanda#polarity returns 5.0 for happy & sademoji
rspec ./spec/sad_panda_spec.rb:75 # SadPanda#polarity returns 8.0 for happy emoji
rspec ./spec/sad_panda/polarity_spec.rb:34 # SadPanda::Polarity#call when input contains a word with polarity returns 10.0
rspec ./spec/sad_panda/polarity_spec.rb:38 # SadPanda::Polarity#call when input contains a word with polarity adds polarities of words to polarities Array```

**Comment by EdwinRozario** [comment]
You are all good mate. It was my bad, i used a Ruby 2.4 syntax `[].sum`. In your case it wasn't working for the ruby version you are running. I have fixed this. And made sure that the specs are passing for even Ruby 1.9.3. I have bumped the version up to 1.1.0 as its not a feature update but a full code base update.

**Comment by mattThousand** [comment]
Great work @EdwinRozario - approved!

**Comment by EdwinRozario** [comment]
👍

**Comment by mattThousand** [comment]
screen shot 2017-10-15 at 9 56 06 pm

**Comment by mattThousand** [comment]
`1.1.0` published on `rubygems`. Many thanks 🚀

**Pull request merged**